libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Functions/MissingPropertiesFunction.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using AdaptiveExpressions;
+using AdaptiveExpressions.Memory;
+using Microsoft.Bot.Builder.Dialogs.Adaptive;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Generators;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Functions
+{
+    /// <summary>
+    /// Defines missingProperties(templateName) expression function.
+    /// </summary>
+    /// <remarks>
+    /// This expression will get all variables the template contains.
+    /// </remarks>
+    public class MissingPropertiesFunction : ExpressionEvaluator
+    {
+        /// <summary>
+        /// Function identifier name.
+        /// </summary>
+        public const string Name = "missingProperties";
+
+        private static DialogContext dialogContext;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MissingPropertiesFunction"/> class.
+        /// </summary>
+        /// <param name="context">Dialog context.</param>
+        public MissingPropertiesFunction(DialogContext context)
+            : base(Name, Function, ReturnType.Array, FunctionUtils.ValidateUnaryString)
+        {
+            dialogContext = context;
+        }
+
+        private static (object value, string error) Function(Expression expression, IMemory state, Options options)
+        {
+            var (args, error) = FunctionUtils.EvaluateChildren(expression, state, options);
+            if (error != null)
+            {
+                return (null, error);
+            }
+
+            var lgm = dialogContext.Services.Get<LanguageGeneratorManager>();
```

Comment:
>dialogContext.Services.Get<LanguageGeneratorManager>(); [](http://example.com/codeflow?start=22&length=55)

You can get access to this through turn.languageGeneratorManager.  That is copied from the dialog context into the turn state on every turn.  This makes it dynamic wrt dialog context.